core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -139,12 +150,16 @@
    @GuardedBy("storesMutex")
    private final List<TransactionalCacheWriter> txWriters = new ArrayList<>();
    private final Semaphore publisherSemaphore = new Semaphore(Integer.MAX_VALUE);
+   // TODO: HAVE to fix this before integrating - move to just semaphore
```

Comment:
Is this outdated?

Reply:
Yes, I was originally hoping to rewrite the locking, but the more I changed things a rewrite wasn't actually required. I can just remove the comment.